crates/ironrdp-rdpdr-native/src/nix/simple.rs

```diff
@@ -0,0 +1,790 @@
+use ironrdp_pdu::PduResult;
+use ironrdp_svc::{impl_as_any, SvcMessage};
+use libc::ftruncate;
+use nix::libc::{closedir, opendir, readdir, DIR};
+
+use ironrdp_rdpdr::pdu::efs::*;
+use ironrdp_rdpdr::pdu::esc::{ScardCall, ScardIoCtlCode};
+use ironrdp_rdpdr::pdu::RdpdrPdu;
+use ironrdp_rdpdr::RdpdrBackend;
+use std::ffi::{CStr, CString};
+use std::io::Read;
+use std::io::{Seek, SeekFrom, Write};
+use std::os::fd::{AsFd, AsRawFd};
+use std::os::unix::fs::MetadataExt;
+
+#[derive(Debug)]
+pub struct SimpleRdpdrBackend {
+    pub file_id: u32,
+    pub file_base: String,
+    pub file_map: std::collections::HashMap<u32, std::fs::File>,
+    pub file_path_map: std::collections::HashMap<u32, String>,
+    pub file_dir_map: std::collections::HashMap<u32, DirBox>,
+}
+
+#[derive(Debug)]
+pub struct DirBox {
```

Comment:
question: Maybe we can replace most of the unsafe code by using [`nix::dir::Dir`](https://docs.rs/nix/latest/nix/dir/struct.Dir.html). Did you consider it?